actors/user/src/main/java/org/sunbird/user/actors/CertificateActor.java

```diff
@@ -17,151 +13,166 @@
+        } else if (ActorOperations.MERGE_USER_CERTIFICATE.getValue().equals(request.getOperation())) {
+            mergeUserCertificate(request);
+        } else{
+            unSupportedMessage();
+        }
     }
-  }
 
-  /**
-   * This method validates the access-code of the certificate and retrieve certificate details.
-   *
-   * @param userRequest
-   */
-  private void getCertificate(Request userRequest) throws IOException {
-    Map request = userRequest.getRequest();
-    String certificatedId = (String) request.get(JsonKey.CERT_ID);
-    String accessCode = (String) request.get(JsonKey.ACCESS_CODE);
-    Map<String, Object> responseDetails = getCertificateDetails(certificatedId);
-    if (responseDetails.get(JsonKey.ACCESS_CODE.toLowerCase()).equals(accessCode)) {
-      Map userResponse = new HashMap<String, Object>();
-      Response userResult = new Response();
-      Map recordStore = (Map<String, Object>) responseDetails.get(JsonKey.STORE);
-      String jsonData = (String) recordStore.get(JsonKey.JSON_DATA);
-      userResponse.put(JsonKey.JSON_DATA, objectMapper.readValue(jsonData, Map.class));
-      userResponse.put(JsonKey.PDF_URL, recordStore.get(JsonKey.PDF_URL));
-      userResponse.put(JsonKey.OTHER_LINK, responseDetails.get(JsonKey.OTHER_LINK));
-      userResult.put(JsonKey.RESPONSE, userResponse);
-      sender().tell(userResult, self());
-    } else {
-      ProjectLogger.log(
-          "CertificateActor:getCertificate: access code is incorrect : " + accessCode,
-          LoggerEnum.ERROR.name());
-      throw new ProjectCommonException(
-          ResponseCode.invalidParameter.getErrorCode(),
-          ProjectUtil.formatMessage(
-              ResponseCode.invalidParameter.getErrorMessage(), JsonKey.ACCESS_CODE),
-          ResponseCode.CLIENT_ERROR.getResponseCode());
+    /** This method merges the certificates from custodian-user to non-custodian-user
+     * @param certRequest
+     */
+    private void mergeUserCertificate(Request certRequest) {
+        Response userResult = new Response();
+        Map request = certRequest.getRequest();
+        String mergeeId = (String) request.get(JsonKey.FROM_ACCOUNT_ID);
+        String mergerId = (String) request.get(JsonKey.TO_ACCOUNT_ID);
+        Response response = cassandraOperation.getRecordsByProperty(certDbInfo.getKeySpace(),certDbInfo.getTableName(),JsonKey.USER_ID,mergeeId);
+        Map<String, Object> record = response.getResult();
+        if (null != record && null != record.get(JsonKey.RESPONSE)) {
+            List responseList = (List) record.get(JsonKey.RESPONSE);
+            if (!responseList.isEmpty()) {
+                responseList.forEach(responseMap -> {
+                    Map<String, Object> responseDetails = (Map<String, Object>) responseMap;
+                    responseDetails.put(JsonKey.USER_ID,mergerId);
+                    cassandraOperation.updateRecord(certDbInfo.getKeySpace(),certDbInfo.getTableName(),responseDetails);
```

Comment:
instead of singlr update we can used bulk update method.

Reply:
The provided method for bulk upload is not working as expected, so proceeding with this.

Reply:
Ok